src/lib/viz/layer/Layer.ts

```diff
@@ -266,7 +266,11 @@ export class Layer extends WithEvents implements StyledLayer {
     if (this._source.sourceType === 'SQLSource' || this._source.sourceType === 'DatasetSource') {
       this._deckLayer = new MVTLayer(layerProperties);
     } else if (this._source.sourceType === 'GeoJSONSource') {
-      this._deckLayer = new GeoJsonLayer(layerProperties);
+      if (layerProperties._isIconLayer) {
+        this._deckLayer = new IconLayer(layerProperties);
```

Comment:
I guess this doesn't have any effect on things related to getting the features and so on, right? For example, have you tried using local dataviews on Layers with IconLayer?

Reply:
Good point! No, I do not

Reply:
it is failing with geojson + icon + dataview

Reply:
then let's fix that! 😄 

After dataviews, interactivity will be the next piece to test!

Reply:
done and checked